core/node/rules/rule_builder_ae.go

```diff
@@ -127,12 +136,12 @@ func (re *ruleBuilderAEImpl) run() (bool, *auth.ChainAuthArgs, *AddEventSideEffe
 	if err != nil {
 		return false, nil, nil, err
 	}
-	if len(re.checks) == 0 && chainAuthArgs == nil && requiredParentEvent == nil {
+	if len(re.checks) == 0 && chainAuthArgsList == nil && requiredParentEvent == nil {
```

Comment:
You probably want to check on `len(chainAuthArgsList) == 0`? If not this check also passed with an empty slice.

Reply:
Yes, you're right - thank you for catching this!